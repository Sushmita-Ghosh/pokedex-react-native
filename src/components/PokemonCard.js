import {
  View,
  Text,
  StyleSheet,
  Image,
  TouchableWithoutFeedback,
} from "react-native";
// import { Image } from "expo-image";
import { images } from "../data/images.js";

import React, { useState, useEffect, memo } from "react";
import { useNavigation } from "@react-navigation/native";
import getColorByPokemonType from "./../utils/getColorByPokemonType";
import { capitalize } from "lodash";
import { getPokemonsApi } from "../api/pokemon";

const PokemonCard = ({ url }) => {
  const [pokemon, setPokemon] = useState();
  const navigation = useNavigation();

  const loadPokemons = async () => {
    const res = await getPokemonsApi(url);
    setPokemon(res);
  };

  useEffect(() => {
    (async () => {
      await loadPokemons();
    })();
  }, []);

  if (!pokemon) return;

  const pokemonColor = getColorByPokemonType(pokemon.types[0].type.name);
  const bgStyles = { backgroundColor: pokemonColor, ...styles.bgStyles };

  const goToPokemonDetailsScreen = () => {
    navigation.navigate("PokemonDetails", {
      id: pokemon.id,
      name: pokemon.name,
    });
  };

  // {
  //   uri: pokemon.sprites.other["official-artwork"].front_default,
  // }
  let imagePokemon =
    images[pokemon.id] !== undefined
      ? images[pokemon.id].image
      : images[1].image;

  return (
    <TouchableWithoutFeedback onPress={goToPokemonDetailsScreen}>
      <View style={styles.spacing}>
        <View style={bgStyles}>
          <Image source={imagePokemon} style={styles.image} />
          <Text style={styles.number}>
            #{`${pokemon.order}`.padStart(3, 0)}
          </Text>
          <Text style={styles.name}>{capitalize(pokemon.name)}</Text>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  spacing: {
    padding: 10,
    margin: 5,
    justifyContent: "space-evenly",
  },
  bgStyles: {
    borderRadius: 15,
    padding: 10,
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    height: 200,
    borderColor: "black",
    borderWidth: 1,
    borderStyle: "dashed",
  },
  image: {
    width: 120,
    height: 120,
  },
});

export default memo(PokemonCard);
