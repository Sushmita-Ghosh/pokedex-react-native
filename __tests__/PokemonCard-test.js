import React from "react";
import {
  render,
  screen,
  fireEvent,
  waitFor,
} from "@testing-library/react-native";
import ShallowRenderer from "react-test-renderer/shallow";
import { NavigationContainer } from "@react-navigation/native";
import PokemonCard from "../src/components/molecules/PokemonCard/PokemonCard.js";

describe("Pokemon Card Component", () => {
  const mockResponse = {
    damage_relations: {
      double_damage_from: [
        {
          name: "fighting",
          url: "https://pokeapi.co/api/v2/type/2/",
        },
      ],
    },
  };

  const mockBaseProps = {
    navigation: {
      navigate: jest.fn(),
    },
  };

  test("Check if Pokemon Card component rendered Correctly", async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({ json: () => Promise.resolve(mockResponse) })
    );
    const renderer = new ShallowRenderer();
    const Component = (props) =>
      renderer.render(<PokemonCard {...mockBaseProps} {...props} />);

    await waitFor(() => {
      expect(Component).toMatchSnapshot();
    });
  });
});
