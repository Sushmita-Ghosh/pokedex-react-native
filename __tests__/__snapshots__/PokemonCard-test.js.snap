// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Card Component Check if Pokemon Card component rendered Correctly 1`] = `[Function]`;
